python/tk_houdini/ui_generation.py

```diff
@@ -447,3 +550,74 @@ def _jump_to_fs(engine):
     engine.launch_command('%s')
 """
 
+# The code that is stored in the python panel interfaces. 
+_g_panel_script = \
+"""
+from PySide import QtGui
+
+class NoPanelWidget(QtGui.QWidget):
+    
+    def __init__(self, msg, error=None):
+    
+        super(NoPanelWidget, self).__init__()
+
+        sg_icon_path = '%s'
+        
+        sg_icon = QtGui.QLabel()
+        sg_icon.setPixmap(QtGui.QPixmap(sg_icon_path))
+        msg_lbl = QtGui.QLabel(msg) 
+        msg_lbl.setWordWrap(True)
+
+        if error:
+            error_txt = QtGui.QTextEdit(error)
+            error_txt.setReadOnly(True)
+
+        h_layout = QtGui.QHBoxLayout()
+        h_layout.setSpacing(5)
+        h_layout.addWidget(sg_icon)
+        h_layout.addWidget(msg_lbl)
+    
+        v_layout = QtGui.QVBoxLayout(self)
+        v_layout.setContentsMargins(10, 10, 10, 10)
+        v_layout.setSpacing(15)
+        v_layout.addStretch()
+        v_layout.addLayout(h_layout)
+        if error:
+            v_layout.addWidget(error_txt)
+        v_layout.addStretch()
+
+def createInterface():
+
+    try:
+        import tank.platform.engine
+    except ImportError:    
+        return NoPanelidget(
+            "It looks like you're running Houdini outside of a Shotgun "
+            "context. Next time you launch Houdini from within a Shotgun "
+           "context, you will see the %s here."
+        )
+    except Exception:
+        import traceback
+        return NoPanelWidget(
+            "There was a problem loading this panel! The error message "
+            "is provided below.",
+            error=traceback.format_exc()
+        )
+    
+    engine = tank.platform.engine.current_engine()
+    panel_info = engine.get_panel_info('%s')
+    panel_widget = panel_info['widget_class']()
+
+    if not panel_widget:
+        panel_widget = NoPanelWidget(
+            "The definition of this panel could not be found." 
```

Comment:
So if they for example have a shot specific panel and they load up an asset config, this message would be displayed? I wonder if we could phrase it to sound less like an error message and more like an info message? Right now it sounds like something went wrong, when that isn't necessarily the case.

Maybe something along the lines of 

- The Shotgun Panel is not available in this configuration
- Asset ABC123 does not have the Shotgun Panel configured

What do you think?